packages/zero-client/src/client/context.ts

```diff
@@ -34,16 +37,29 @@ export class ZeroContext implements QueryDelegate {
   readonly #batchViewUpdates: (applyViewUpdates: () => void) => void;
   readonly #commitListeners: Set<CommitListener> = new Set();
 
+  readonly slowMaterializationTheshold: number | undefined;
   readonly staticQueryParameters = undefined;
 
   constructor(
+    lc: LogContext,
     mainSources: IVMSourceBranch,
     addQuery: AddQuery,
     batchViewUpdates: (applyViewUpdates: () => void) => void,
+    slowMaterializationTheshold: number | undefined,
   ) {
+    this.#lc = lc;
     this.#mainSources = mainSources;
     this.#addQuery = addQuery;
     this.#batchViewUpdates = batchViewUpdates;
+    this.slowMaterializationTheshold = slowMaterializationTheshold;
+  }
+
+  get lc() {
```

Comment:
Do we really need to make this part of the interface?

Reply:
I tried it by alternately passing the lc into `QueryImpl` and it's a massive yak shave to pass it through everywhere.